### PR TITLE
CLOUDP-128166: Add all flags except "policy" to the command

### PR DESCRIPTION
**Pull request by Razeer123**
## Proposed changes

Adds all flags except "policy" one to the base version of the "atlas backups schedule update" command.

**This PR depends on: https://github.com/mongodb/mongodb-atlas-cli/pull/1368**

_Jira ticket:_ [CLOUDP-128166](https://jira.mongodb.org/browse/CLOUDP-128166)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/backup/schedule/update.go [hunk 1]
```diff
@@ -43,35 +56,168 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
...
 		Short:   "Update a snapshot backup policies for a cluster.",
+		Example: fmt.Sprintf(`  The following updates a snapshot backup policies for a cluster Cluster0:
+  $ %s backup schedule update --clusterName Cluster0 --updateSnapshots --exportBucketId 62c569f85b7a381c093cc539 --exportFrequencyType monthly`, cli.ExampleAtlasEntryPoint()),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			preRun := opts.PreRunE(
 				opts.ValidateProjectID,
 				opts.initStore(cmd.Context()),
 				opts.InitOutput(cmd.OutOrStdout(), updateTemplate),
 			)
+			err := opts.verifyExportFrequencyType()
```
Comment (fmenezes) [inline internal/cli/atlas/backup/schedule/update.go:179]: How about using `opts.PreRunE(..., opts.verifyExportFrequencyType())`? It enforces all validations currently.

Comment (Razeer123) [reply internal/cli/atlas/backup/schedule/update.go:179]: @fmenezes Makes more sense that my implementation. I will update the code.

File: internal/cli/atlas/backup/schedule/update.go [hunk 2]
```diff
@@ -43,35 +56,163 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
...
+		if cmd.Flags().Changed(flag.RestoreWindowDays) {
+			if opts.restoreWindowDays <= 0 {
+				return errors.New("incorrect value for parameter restoreWindowDays. Value must be a positive, non-zero integer")
+			}
+		}
+		return nil
+	}
+}
+
+func checkForExport(out *atlas.CloudProviderSnapshotBackupPolicy) {
```
[the remark this answers is not in the document]
Comment (Razeer123) [reply internal/cli/atlas/backup/schedule/update.go:151]: @fmenezes Without checking if `Export` is allocated, we'll get the segmentation fault (SIGSEGV).

[the remark this answers is not in the document]
Comment (fmenezes) [reply internal/cli/atlas/backup/schedule/update.go:151]: [q] would it be able to be always present?

```
out := &atlas.CloudProviderSnapshotBackupPolicy{Export:&atlas.Export{}}
```

[the remark this answers is not in the document]
Comment (Razeer123) [reply internal/cli/atlas/backup/schedule/update.go:151]: @fmenezes I don't think so. If it's always initialised and we want to perform an update that doesn't involve the `Export` struct, for some reason we'll get an internal server error.
```
michal.derej@M-L7H6RDJWQX mongodb-atlas-cli % ./bin/atlas backup schedule update --clusterName Cluster0 --noUpdateSnapshots -P dev                                                                              
Error: PATCH https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/62ab4d9e22f63b08ef5876f7/clusters/Cluster0/backup/schedule: 500 (request "UNEXPECTED_ERROR") Unexpected error.
michal.derej@M-L7H6RDJWQX mongodb-atlas-cli % ./bin/atlas backup schedule update --clusterName Cluster0 --referenceHourOfDay 22 -P dev                                                              
Error: PATCH https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/62ab4d9e22f63b08ef5876f7/clusters/Cluster0/backup/schedule: 500 (request "UNEXPECTED_ERROR") Unexpected error.
```

File: internal/cli/atlas/backup/schedule/update.go [hunk 3]
```diff
@@ -43,35 +56,163 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
...
+	}
+
+	out.AutoExportEnabled = returnValueForSetting(opts.autoExport, opts.noAutoExport)
+	out.UpdateSnapshots = returnValueForSetting(opts.updateSnapshots, opts.noUpdateSnapshots)
+	out.UseOrgAndGroupNamesInExportPrefix = returnValueForSetting(opts.useOrgAndGroupNamesInExportPrefix, opts.noUseOrgAndGroupNamesInExportPrefix)
+
+	return out, nil
+}
+
+func (opts *UpdateOpts) verifyExportBucketID(out *atlas.CloudProviderSnapshotBackupPolicy) {
```
Comment (fmenezes) [inline internal/cli/atlas/backup/schedule/update.go:100]: seems that we use verify for validations and now verify for assigning, how about moving this code to `NewBackupConfig` ?

File: internal/cli/atlas/backup/schedule/update.go [hunk 4]
```diff
@@ -43,35 +56,163 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
...
+	}
+}
+
+func checkForExport(out *atlas.CloudProviderSnapshotBackupPolicy) {
+	if out.Export == nil {
+		out.Export = new(atlas.Export)
+	}
+}
+
+func returnValueForSetting(enableFlag, disableFlag bool) *bool {
```
Comment (fmenezes) [inline internal/cli/atlas/backup/schedule/update.go:157]: I think I've seen it before no?
can we reuse it across packages?

File: internal/cli/atlas/backup/schedule/update.go (hunk 3 above)
Comment (Razeer123) [reply internal/cli/atlas/backup/schedule/update.go:100]: Good idea. Name of this function is misleading.

File: internal/cli/atlas/backup/schedule/update.go (hunk 4 above)
Comment (Razeer123) [reply internal/cli/atlas/backup/schedule/update.go]: Sure! I moved it to another package.

**Review by fmenezes — APPROVED** [review]

**Pull request merged**